server/events/apply_command_runner.go

```diff
@@ -2,14 +2,49 @@ package events
 
 import (
 	"github.com/runatlantis/atlantis/server/events/db"
+	"github.com/runatlantis/atlantis/server/events/locking"
 	"github.com/runatlantis/atlantis/server/events/models"
 	"github.com/runatlantis/atlantis/server/events/vcs"
 )
 
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_apply_command_locker.go ApplyCommandLocker
+
+type ApplyCommandLocker interface {
+	IsDisabled(ctx *CommandContext) bool
+}
+
+func NewApplyCommandLocker(
+	applyLockChecker locking.ApplyLockChecker,
+	disableApply bool,
```

Comment:
can we rename this to disableApplyFlag or something that hints at this being a global server flag?